client/state/initial-state.js

```diff
@@ -108,10 +109,18 @@ export async function getStateFromLocalStorage( reducer, subkey ) {
 	const reduxStateKey = getReduxStateKey() + ( subkey ? ':' + subkey : '' );
 
 	try {
-		const storedState = await getStoredItem( reduxStateKey );
+		let loggedOutState;
+		let storedState = await getStoredItem( reduxStateKey );
+		if ( reduxStateKey !== 'redux-state-logged-out' ) {
+			loggedOutState = await getStoredItem( 'redux-state-logged-out' );
```

Comment:
One more thing: is it only the `state.signup` subtree that we want to transfer from logged-out to logged-in state? Then I would pick it from the logged-out state explicitly and throw the rest away. That's a precaution against spreading garbage data 🙂 

One more idea how to transfer only the `signup` state:
1. Wrap the `signup` reducer with `withStorageKey` wrapper. That will cause it to persist to a separate storage key, `redux-state-123:signup`, instead of the general `redux-state-123`.
2. Move the fallback-to-logged-out logic to the `loadReducerState` function that fetches all these sub-states from storage. Then we can enable the fallback only for `signup` key.

That's another way, slightly better, to limit the fallback only to the `signup` reducer.

Reply:
Thanks for guiding me here @jsnajdr, that's a much better idea!